crates/libeq_wld/src/parser/mod.rs

```diff
@@ -245,147 +296,226 @@ impl<'a> FragmentHeader<'a> {
     }
 
     fn parse_body(&self) -> Box<FragmentType> {
+        let debug_msg = &format!("{:02x?}", self);
         match self.fragment_type {
             AlternateMeshFragment::TYPE_ID => Box::new(FragmentType::AlternateMesh(
-                AlternateMeshFragment::parse(&self.field_data).unwrap().1,
+                AlternateMeshFragment::parse(&self.field_data)
+                    .expect(debug_msg)
+                    .1,
```

Comment:
Some (slightly) improved debugging around parse failures. `nom` isn't great at this. And every time I try to understand more detailed error handling in nom I walk away frustrated.